app/src/main/java/com/jHerscu/clearskies/ui/prefs/PreferencesViewModel.kt

```diff
@@ -1,8 +1,117 @@
 package com.jHerscu.clearskies.ui.prefs
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.jHerscu.clearskies.data.repo.PreferencesRepoImpl
+import com.jHerscu.clearskies.di.DefaultDispatcher
+import com.jHerscu.clearskies.domain.repoInterface.PreferenceGroup
+import com.jHerscu.clearskies.domain.repoInterface.PreferenceOrderComparator
+import com.jHerscu.clearskies.domain.repoInterface.TranslatedPrefGroup
+import com.jHerscu.clearskies.domain.repoInterface.UserPrefs
+import com.jHerscu.clearskies.domain.useCase.preferences.ReadUserPrefsUseCase
+import com.jHerscu.clearskies.utils.FLOW_SUBSCRIPTION_STOP_TIMEOUT_MILLIS
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combineTransform
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 @HiltViewModel
-class PreferencesViewModel @Inject constructor() : ViewModel()
+class PreferencesViewModel @Inject constructor(
+    private val preferencesRepo: PreferencesRepoImpl, // TODO(jherscu): Decide on repo impl pattern
+    readUserPrefs: ReadUserPrefsUseCase,
+    @DefaultDispatcher private val defaultDispatcher: CoroutineDispatcher
+) : ViewModel() {
+
+    private val _prefState = readUserPrefs()
+    private val _prefGroups = MutableStateFlow(PreferenceGroup.defaultSortOrder)
+
+    val viewState: StateFlow<ViewState> = _prefGroups
+        .combineTransform(_prefState) { prefGroupOrder, userPrefs ->
+            emit(ViewState(userPrefs = userPrefs, prefGroups = prefGroupOrder))
+        }.stateIn(
+            scope = viewModelScope,
+            started = SharingStarted.WhileSubscribed(FLOW_SUBSCRIPTION_STOP_TIMEOUT_MILLIS),
+            initialValue = ViewState()
+        )
```

Comment:
Uses prefs as source of truth for pref screen view state, with exception of sort list

Reply:
Replaced with more natural integration of updating `prefState` in `_viewState` via collectLatest in init block. Does the same thing, but more concise and readable. Still uses readUserPrefs as source of truth for screen